zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/JobControllerTest.java

```diff
@@ -337,4 +337,161 @@ void shouldCompleteJobWithVariables() {
 
     Mockito.verify(jobServices).completeJob(1L, Map.of("foo", "bar"));
   }
+
+  @Test
+  void shouldUpdateJob() {
+    // given
+    when(jobServices.updateJob(anyLong(), anyInt(), anyLong()))
+        .thenReturn(CompletableFuture.completedFuture(new JobRecord()));
+
+    final var request =
+        """
+        {
+          "changeset": {
+            "retries": 5,
+            "timeout": 1000
+          }
+        }""";
+    // when/then
+    webClient
+        .patch()
+        .uri(JOBS_BASE_URL + "/1")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isNoContent();
+
+    Mockito.verify(jobServices).updateJob(1L, 5, 1000L);
+  }
+
+  @Test
+  void shouldUpdateJobWithOnlyRetries() {
+    // given
+    when(jobServices.updateJob(anyLong(), anyInt(), anyLong()))
+        .thenReturn(CompletableFuture.completedFuture(new JobRecord()));
+
+    final var request =
+        """
+        {
+          "changeset": {
+            "retries": 5
+          }
+        }""";
+    // when/then
+    webClient
+        .patch()
+        .uri(JOBS_BASE_URL + "/1")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isNoContent();
+
+    Mockito.verify(jobServices).updateJob(1L, 5, 0L);
+  }
+
+  @Test
+  void shouldUpdateJobWithOnlyTimeout() {
+    // given
+    when(jobServices.updateJob(anyLong(), anyInt(), anyLong()))
+        .thenReturn(CompletableFuture.completedFuture(new JobRecord()));
+
+    final var request =
+        """
+        {
+          "changeset": {
+            "timeout": 1000
+          }
+        }""";
+    // when/then
+    webClient
+        .patch()
+        .uri(JOBS_BASE_URL + "/1")
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isNoContent();
+
+    Mockito.verify(jobServices).updateJob(1L, 0, 1000L);
+  }
+
+  @Test
+  void shouldRejectUpdateJob() {
```

Comment:
🔧 We can also test for a missing request body additionally.

Reply:
Same reply as the other PR. We are going to test the framework behaviour. But I think that in this specific case it make sense 👍